packages/core/src/animation/Animator.ts

```diff
@@ -0,0 +1,463 @@
+import { AnimatorUtils } from "./AnimatorUtils";
+import { AnimateProperty } from "./enums/AnimateProperty";
+import { AnimatorLayerBlendingMode } from "./enums/AnimatorLayerBlendingMode";
+import { PlayType } from "./enums/PlayType";
+
+/**
+ * The controller of the animation system.
+ */
+export class Animator extends Component {
+  /** The mode of the Animator recorder. */
+  recorderMode: AnimatorRecorderMode = AnimatorRecorderMode.Offline;
+  /** The playback speed of the Animator. 1 is normal playback speed. */
+  speed: number = 1;
+
+  private _animatorController: AnimatorController;
+  private _diffValueFromBasePos: InterpolableValue;
+  private _diffFloatFromBasePos: number = 0;
+  private _diffVector2FromBasePos: Vector2 = new Vector2();
+  private _diffVector3FromBasePos: Vector3 = new Vector3();
+  private _diffVector4FromBasePos: Vector4 = new Vector4();
+  private _diffQuaternionFromBasePos: Quaternion = new Quaternion();
+  private _tempVector3: Vector3 = new Vector3();
+  private _tempQuaternion: Quaternion = new Quaternion();
+
+  /**
+   * Get the AnimatorController that controls the Animator.
+   */
+  get animatorController(): AnimatorController {
+    return this._animatorController;
+  }
+
+  /**
+   * Set the AnimatorController that controls the Animator.
+   */
+  set animatorController(animatorController: AnimatorController) {
+    this._animatorController = animatorController;
+    if (!animatorController) return;
+    animatorController._setTarget(this.entity);
+  }
+
+  /**
+   * Get all layers from the AnimatorController which belongs this Animator .
+   */
+  get layers(): Readonly<AnimatorControllerLayer[]> {
+    return this._animatorController?.layers || [];
+  }
+
+  /**
+   * @param entity - The entitiy which the animator component belongs to
+   */
+  constructor(entity: Entity) {
+    super(entity);
+  }
+
+  /**
+   * Plays a state by name.
+   * @param stateName - The state name
+   * @param layerIndex - The layer index(default 0)
+   * @param normalizedTime - The time offset between 0 and 1(default 0)
+   */
+  playState(stateName: string, layerIndex: number = 0, normalizedTime: number = 0): void {
+    const { animatorController } = this;
+    if (!animatorController) return;
+    const animLayer = animatorController.layers[layerIndex];
+    const theState = animLayer.stateMachine.findStateByName(stateName);
+    theState.frameTime = theState.clip.length * normalizedTime;
+    animLayer._playingState = theState;
+    this.recorderMode = AnimatorRecorderMode.Playback;
+  }
+
+  /**
+   * Set the Animator in playback mode.
+   */
+  play(): void {
+    this.recorderMode = AnimatorRecorderMode.Playback;
+  }
+
+  /**
+   * Stops the animator playback mode.
+   */
+  stop(): void {
+    this.recorderMode = AnimatorRecorderMode.Offline;
+  }
+
+  /**
+   * Evaluates the animator component based on deltaTime.
+   * @param deltaTime - The deltaTime when the animation update
+   * @private
```

Comment:
this function is public? If public delete `@private`,if not add  `_` for function name.

Reply:
done